crates/wkg/src/oci.rs

```diff
@@ -129,6 +130,24 @@ pub struct PushArgs {
 
     /// The path to the file to push
     pub file: PathBuf,
+
+    /// Add an OCI annotation to the image manifest
+    #[clap(long = "annotation", value_parser = parse_key_val::<String, String>)]
+    pub annotation: Vec<(String, String)>,
+}
+
+/// Parse a single key-value pair
+fn parse_key_val<T, U>(s: &str) -> Result<(T, U), Box<dyn Error + Send + Sync + 'static>>
```

Comment:
Any reason this needs to be generic? We'll probably only use this to get two strings out, so it could be simpler just to omit the generics and the list of bounds. It also means we don't have to parse and instead can just to_owned the split string.

Reply:
Nit: Also, I am pretty sure this can just be `anyhow::Error` since we use anyhow most places here.

Reply:
Heh, sure I can change it to not be generic. This should honestly probably be a `clap` built-in, since I copied this specific function from their examples since it's apparently a common case.